### DISCUSSION TITLE
Sort project config YAML UIDs?

### DISCUSSION THREAD

**Discussion by mildlygeeky** [craftcms/cms#9144] [Ideas] [answered]
I am finding diff's of project config changes (particularly after an upgrade where the files end up being rewritten) have the array keys (often UIDs) being shuffled around, making it hard to tell meaningful changes from something like the below (below nothing has actually changed - just the order of the UIDs):

<img width="700" alt="Screen Shot 2021-07-08 at 10 58 37 PM" src="https://user-images.githubusercontent.com/273266/125017014-4e3bc200-e040-11eb-92e4-232a8bc8db95.png">

Not sure how/if it is possible, but would love to see some sort of standard sorting be enforced here (alphabetical)? Currently working through what looks like 300+ changed project config files with a 3.6 upgrade, but most are just reshuffling of UIDs.

**Comment by brandonkelly** [comment, marked as answer]
Everything already gets sorted. Guessing the reason you’re seeing this is because we started sorting using `SORT_NATURAL` in 3.6.11 (4d87169c503dbb69fe26219f7fbb82026e8a7220), fixing some edge case bugs such as https://github.com/presseddigital/linkit/issues/88. The left side of your diff is what you’d get with `SORT_REGULAR`, whereas the right side is what you’d get with `SORT_NATURAL`.

**Reply by mildlygeeky** [reply]
Aaah, that makes sense, and LinkIt was definitely a place where we'd seen issues in the past, so glad to hear that `SORT_NATURAL` should put that behind us. Thanks!
